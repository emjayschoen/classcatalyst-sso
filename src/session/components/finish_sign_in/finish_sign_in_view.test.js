import React from 'react';
import { shallow } from 'enzyme';
import FinishSignInView from './finish_sign_in_view';

const NO_SEARCH = {};
const LEGIT_SEARCH = { search: `?fork=spoon&returnTo=${encodeURIComponent('/foo/bar')}&nothing=something` };
const DUMB_SEARCH = { search: '?why=does&this=confusion' };

describe('<FinishSignInView />', () => {
  let subject;

  describe('when not signed in', () => {
    describe('without search params', () => {
      beforeEach(() => {
        subject = shallow(<FinishSignInView location={NO_SEARCH} signedIn={false} finishSignIn={jest.fn()} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('with meaningless search params', () => {
      beforeEach(() => {
        subject = shallow(<FinishSignInView location={DUMB_SEARCH} signedIn={false} finishSignIn={jest.fn()} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('with returnTo search params', () => {
      beforeEach(() => {
        subject = shallow(<FinishSignInView location={LEGIT_SEARCH} signedIn={false} finishSignIn={jest.fn()} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });

  describe('when signed in', () => {
    describe('without search params', () => {
      beforeEach(() => {
        subject = shallow(<FinishSignInView location={NO_SEARCH} signedIn finishSignIn={jest.fn()} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('with meaningless search params', () => {
      beforeEach(() => {
        subject = shallow(<FinishSignInView location={DUMB_SEARCH} signedIn finishSignIn={jest.fn()} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('with returnTo search params', () => {
      beforeEach(() => {
        subject = shallow(<FinishSignInView location={LEGIT_SEARCH} signedIn finishSignIn={jest.fn()} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});
