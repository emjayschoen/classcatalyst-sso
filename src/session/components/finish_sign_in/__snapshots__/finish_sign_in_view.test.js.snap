// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinishSignInView /> when not signed in with meaningless search params renders correctly 1`] = `
<styled.div>
  <Spinner />
</styled.div>
`;

exports[`<FinishSignInView /> when not signed in with returnTo search params renders correctly 1`] = `
<styled.div>
  <Spinner />
</styled.div>
`;

exports[`<FinishSignInView /> when not signed in without search params renders correctly 1`] = `
<styled.div>
  <Spinner />
</styled.div>
`;

exports[`<FinishSignInView /> when signed in with meaningless search params renders correctly 1`] = `
<Redirect
  to={
    Object {
      "pathname": "/signin",
      "state": Object {
        "returnTo": undefined,
      },
    }
  }
/>
`;

exports[`<FinishSignInView /> when signed in with returnTo search params renders correctly 1`] = `
<Redirect
  to={
    Object {
      "pathname": "/signin",
      "state": Object {
        "returnTo": "/foo/bar",
      },
    }
  }
/>
`;

exports[`<FinishSignInView /> when signed in without search params renders correctly 1`] = `
<Redirect
  to={
    Object {
      "pathname": "/signin",
      "state": Object {
        "returnTo": undefined,
      },
    }
  }
/>
`;
